packages/vscode-extension/lib/expo/expo_start.js

```diff
@@ -4,15 +4,65 @@ const {
   overrideModuleFromAppDependency,
 } = require("../metro_helpers");
 
+const { gte } = require("../third-party/semver");
+
+const EXPO_CLI_VERSION = requireFromAppDependency("expo", "@expo/cli/package.json").version;
+
+const MINIMUM_RESOLVE_CONFIG_OVERRIDE_VERSION = "54.0.20";
+
+function requireGetDefaultConfig() {
+  try {
+    const metroConfig = requireFromAppDependency("expo", "@expo/metro-config");
+    return metroConfig.getDefaultConfig
+  } catch {
+    // metro-config is only available on newer expo versions, 
+    // so we just fall back to undefined. Note that this path
+    // should be unreachable as we check for expo version before 
+    // ever using getDefaultConfig. 
+    return undefined;
+  }
+}
+
+function createRadonLoadConfig(original) {
+  return async (...args) => {
+    const config = await original(...args);
+    return adaptMetroConfig(config);
+  }
+}
+
+function createRadonResolveConfig(original) {
+  return async (expoConfig, projectRoot, ...rest) => {
+    const originalResolvedConfig = await original(expoConfig, projectRoot, ...rest);
+    const getDefaultConfig = requireGetDefaultConfig();
+    if (getDefaultConfig === undefined) {
+      return originalResolvedConfig;
+    }
```

Comment:
I wonder if we could just use empty config  here instead of loading the default one. I think that the resolved config is later merged with the default anyway, so maybe we can simplify the logic here and also avoid loading the default twice